backend/corpora/lambdas/api/v1/curation/collections/common.py

```diff
@@ -230,20 +230,17 @@ def list_collections_curation(
     return resp_collections
 
 
-@staticmethod
-def add_collection_level_processing_status(collection: DbCollection):
+def add_collection_level_processing_status(collection: DbCollection) -> str:
     # Add a Collection-level processing status
-    status = None
-    has_statuses = False
+    if not collection.datasets:  # Return None if the collection has no datasets.
+        return None
+    return_status = ProcessingStatus.SUCCESS
     for dataset in collection.datasets:
-        processing_status = dataset.processing_status
-        if processing_status:
-            has_statuses = True
-            if processing_status.processing_status == ProcessingStatus.PENDING:
-                status = ProcessingStatus.PENDING
-            elif processing_status.processing_status == ProcessingStatus.FAILURE:
-                status = ProcessingStatus.FAILURE
+        if processing_status := dataset.processing_status:
+            status = processing_status.processing_status
+            if status == ProcessingStatus.PENDING:
+                return_status = status
+            elif status == ProcessingStatus.FAILURE:
+                return_status = status
                 break
```

Comment:
```suggestion
                return status
```

Reply:
Prevents accidents in the case where some code doing who-knows-what later gets inserted between the break and the final `return`.